test/functional/wallet_migration.py

```diff
@@ -299,6 +299,40 @@ def test_multisig(self):
         assert_equal(ms1_solvable.getbalance(), 0)
         assert_equal(ms1_solvable.listtransactions(), [])
 
+    def test_multisig_invalid(self):
+        self.log.info("Test migration of a legacy-wise non-standard bare multisig")
+        wallet = self.create_legacy_wallet("multi_nonstandard")
+
+        # Create enough keys for all coming tests
+        addys = [wallet.getnewaddress()] * 20
+        pubkeys = []
+        privkeys = []
+        for addr in addys:
+            pubkeys.append(wallet.getaddressinfo(addr)['pubkey'])
+            privkeys.append(wallet.dumpprivkey(addr))
+
+        # Create a non-standard multi(4, keys)
+        res = wallet.createmultisig(4, pubkeys[:4])
+        # Import script as a bare multisig. This is standard-wise non-spendable, and it is not allowed descriptors' wise
+        wallet.importaddress(address=res['redeemScript'])
+
+        # Now migrate it and verify we don't crash due to a non-allowed descriptor migration
+        wallet.migratewallet()
+        wallet.unloadwallet()
+
+        ##############################################################
+        # Import a consensus-wise invalid p2sh multisig with 20 keys #
```

Comment:
In d64193142792e68636bf61ef4b083dfb8c1d20d7: A p2sh multisig with 20 keys is invalid? I thought 20 keys are the limit, but still valid.

Reply:
> In [d641931](https://github.com/bitcoin/bitcoin/commit/d64193142792e68636bf61ef4b083dfb8c1d20d7): A p2sh multisig with 20 keys is invalid? I thought 20 keys are the limit, but still valid.

Yes, they are. The limit is not based on the number of keys, it is on the size of the redeem script. Which can go up to 520 bytes max (which limits the number of keys inside a p2sh to 15).